services/ui-src/src/components/cards/EntityCard/EntityCard.tsx

```diff
@@ -83,11 +92,15 @@ export const EntityCard = ({
             data-testid="print-status-indicator"
           >
             <Image
-              src={entityCompleted ? completedIcon : unfinishedIcon}
+              src={
+                entityCompleted && reportingPeriodCompletedOrOptional
+                  ? completedIcon
+                  : unfinishedIcon
+              }
               alt={`entity is ${entityCompleted ? "complete" : "incomplete"}`}
               sx={sx.printVersionIcon}
             />
-            {entityCompleted ? (
+            {entityCompleted && reportingPeriodCompletedOrOptional ? (
```

Comment:
suggestion: since `entityCompleted && reportingPeriodCompletedOrOptional` is used three times we could write it as a variable in this file for simplicity